### PR TITLE
[Bugfix] Add coverage check on atlas exports

### LINKED ISSUE #26144: [Layout][Atlas]Disable the atlas tools when no coverage layer is provided or give more hints on the error message
Author Name: **Harrissou Santanna** (@DelazJ)
Original Redmine Issue: [18253](https://issues.qgis.org/issues/18253)
Affected QGIS version: 3.1(master)
Redmine category:map_composer/printing

---
Enable "atlas generation" in a new print composer
and without selecting a coverage layer launch the export to image (or anything else).
The process of folder selection goes normall and at the end you get an error message: *Error encountered while exporting PDF*. Nothing more.
Given that none of the atlas tools is usable as long as no coverage layer is selected, either we disable those tools or the error message points the user to where the issue resides.

### PR DESCRIPTION
**Pull request by roya0045**
Fixes #26144

## Description

Add an error message when trying to export an atlas without a coverage layer.

### CODE REVIEW

**Comment by m-kuhn** [comment]
Did you manually test this?

**Comment by roya0045** [comment]
@m-kuhn 🤦‍♂ forgot to add negation and just checked that all tests passed. Tested, all is fine now.

**Comment by nyalldawson** [comment]
@roya0045 
> forgot to add negation and just checked that all tests passed.

PLEASE - you need to actually build and test your changes. This development practice of firing off untested work is seriously dangerous and damages the credibility of your other PRs. At this stage I have to pick through them line by line and basically re-implement them again myself, because I just don't know if you've even built your work and tested it locally. It's the reason your PRs keep getting stuck in limbo -- we just don't have the time (or mental energy) to review PRs when the original author hasn't done their own due-diligence.

PLEASE PLEASE PLEASE know that your efforts are extremely valuable. You're doing fantastic work triaging bug reports and submitting fixes like this, and the last thing I want to do is discourage you from this! it's **really** appreciated, and your contributions ARE having a extremely beneficial impact on the QGIS project. We've just gotta get your PR/coding approach smoothed out and then it'll be full steam ahead! :smile:

File: src/app/layout/qgslayoutdesignerdialog.cpp [hunk 1]
```diff
@@ -2808,6 +2817,15 @@ void QgsLayoutDesignerDialog::exportAtlasToSvg()
   if ( !printAtlas || !printAtlas->enabled() )
     return;
 
+  if ( !printAtlas->coverageLayer() )
+  {
+    QMessageBox::warning( this, tr( "Export Atlas as Image" ),
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.cpp]:
```suggestion
    QMessageBox::warning( this, tr( "Export Atlas" ),
```

File: src/app/layout/qgslayoutdesignerdialog.cpp [hunk 2]
```diff
@@ -2976,6 +2994,15 @@ void QgsLayoutDesignerDialog::exportAtlasToPdf()
...
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.cpp]:
```suggestion
    QMessageBox::warning( this, tr( "Export Atlas" ),
```

File: src/app/layout/qgslayoutdesignerdialog.cpp [hunk 3]
```diff
@@ -2651,6 +2651,15 @@ void QgsLayoutDesignerDialog::exportAtlasToRaster()
...
```
Comment (nyalldawson) [inline src/app/layout/qgslayoutdesignerdialog.cpp]:
```suggestion
    QMessageBox::warning( this, tr( "Export Atlas" ),
```

**Comment by roya0045** [comment]
> @roya0045
> 
> > forgot to add negation and just checked that all tests passed.
> 
> PLEASE - you need to actually build and test your changes. This development practice of firing off untested work is seriously dangerous and damages the credibility of your other PRs. At this stage I have to pick through them line by line and basically re-implement them again myself, because I just don't know if you've even built your work and tested it locally. It's the reason your PRs keep getting stuck in limbo -- we just don't have the time (or mental energy) to review PRs when the original author hasn't done their own due-diligence.
> 
> PLEASE PLEASE PLEASE know that your efforts are extremely valuable. You're doing fantastic work triaging bug reports and submitting fixes like this, and the last thing I want to do is discourage you from this! it's **really** appreciated, and your contributions ARE having a extremely beneficial impact on the QGIS project. We've just gotta get your PR/coding approach smoothed out and then it'll be full steam ahead! 😄

Big PRs are tested these days, sometimes with quickfixes that seem trivial I just have a tendency to make the PR as soon as the branch is building properly. I'll try to stave off in the future and take the time to test even minimal changes from now on.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**